### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by Eternal-Flame085**
- Created Dashboard spec
- Created dashboard route
- Created controller
- Created Model methods
- Added model spec test

- Still needs to be updated with friends table

### CODE REVIEW

File: spec/features/dashboard/dashboard_spec.rb [hunk 1]
```diff
@@ -21,4 +21,33 @@
...
+    end
+  end
+
+  it "User currently has no friends" do
+    within id="#Friends" do
+      expect(page).to have_content('You currently have no friends')
+    end
+  end
+
+  it "Section show current parties and thir info for user if any" do
```
Comment (stillsheryl) [inline spec/features/dashboard/dashboard_spec.rb]: typo here: their

File: spec/features/dashboard/dashboard_spec.rb [hunk 2]
```diff
@@ -21,4 +21,33 @@
...
+    current_user = User.last
+    movie = Movie.create!(title: 'Avengers', api_id: 550)
+    party = Party.create!(user_id: current_user.id, movie_id: movie.id, date: '1/12/2020', time: '2:00 PM', duration: 220)
```
Comment (stillsheryl) [inline spec/features/dashboard/dashboard_spec.rb]: We can double check the date when we create the form to make a new party.

File: spec/features/dashboard/dashboard_spec.rb [hunk 3]
```diff
@@ -21,4 +21,33 @@
...
+
+    visit '/dashboard'
+    
+    within id="#party-#{party.id}" do
+      expect(page).to have_content('Avengers')
+      expect(page).to have_content('December 1, 2020')
+      expect(page).to have_content('2:00 PM')
+      expect(page).to have_content('Hosting')
+    end
+  end
```
Comment (stillsheryl) [inline spec/features/dashboard/dashboard_spec.rb:53]: We might want to add an edge case to test for a guest as well as a host?

**Review by stillsheryl — APPROVED** [review]
Looks good to me

**Review by Stearnzy — APPROVED** [review]

**Review by Stearnzy — APPROVED** [review]
